feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/OrchestratorActivity.kt

```diff
@@ -5,36 +5,50 @@ import android.os.Bundle
 import androidx.core.os.bundleOf
 import androidx.navigation.findNavController
 import com.simprints.core.tools.activity.BaseActivity
+import com.simprints.feature.orchestrator.cache.OrchestratorCache
 import com.simprints.feature.orchestrator.databinding.ActivityOrchestratorBinding
 import com.simprints.infra.orchestration.data.results.AppResult
 import com.simprints.infra.uibase.navigation.handleResult
 import com.simprints.infra.uibase.viewbinding.viewBinding
 import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
 
 @AndroidEntryPoint
 internal class OrchestratorActivity : BaseActivity() {
 
     private val binding by viewBinding(ActivityOrchestratorBinding::inflate)
 
+    @Inject
+    lateinit var orchestratorCache: OrchestratorCache
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(binding.root)
 
-        binding.orchestrationHost.handleResult<AppResult>(this, R.id.orchestratorRootFragment) { result ->
+        binding.orchestrationHost.handleResult<AppResult>(
+            this,
+            R.id.orchestratorRootFragment
+        ) { result ->
             setResult(result.resultCode, Intent().putExtras(result.extras))
+
+            orchestratorCache.isExecuting.set(false)
             finish()
         }
     }
 
     override fun onStart() {
         super.onStart()
-        val action = intent.action.orEmpty()
-        val extras = intent.extras ?: bundleOf()
-
-        findNavController(R.id.orchestrationHost).setGraph(
-            R.navigation.graph_orchestration,
-            OrchestratorFragmentArgs(action, extras).toBundle()
-        )
+        if (orchestratorCache.isExecuting.compareAndSet(false, true))  {
+            val action = intent.action.orEmpty()
+            val extras = intent.extras ?: bundleOf()
+
+            findNavController(R.id.orchestrationHost).setGraph(
+                R.navigation.graph_orchestration,
+                OrchestratorFragmentArgs(action, extras).toBundle()
+            )
+        } else {
+            finish()
```

Comment:
Maybe we should add a Simber.e() here to track how often this happens!?

Reply:
Yeah, I had a message in a different spot and forgot to add it here when changing the solution.